### PR TITLE
Add password reset

### PR DESCRIPTION
**Pull request by kiprot**
- add a func to reset password for users who forget it
 - users will get an email from firebase prompting them to enter and set their new password

[new change]
- implemented recommended suggestion to show "forgot password" if a user enters wrong password

### CODE REVIEW

File: src/main/ngapp/src/app/login/login.component.html [hunk 1]
```diff
@@ -44,6 +44,15 @@
...
     </div>
     <br />
+
+    <div class="register_link">
+      Forgot Password? 
+      <a routerLink="" (click)="sendPasswordResetEmail(userEmail.value)">
+        Get email to reset
+      </a>
+    </div>
+    <br />
```
Comment (MohamedShatry) [inline src/main/ngapp/src/app/login/login.component.html:54]: Can we hide this until a user inputs a wrong password?

Comment (kiprot) [reply src/main/ngapp/src/app/login/login.component.html:54]: alright
on it

Comment (kiprot) [reply src/main/ngapp/src/app/login/login.component.html]: implemented

**Comment by kiprot** [comment]
[password_reset_screencast.zip](https://github.com/googleinterns/step147-2020/files/5031820/password_reset_screencast.zip)

- github does not support videos, but this is a downloadable zip that shows the screencast

File: src/main/ngapp/src/app/auth/auth.service.ts [hunk 2]
```diff
@@ -105,6 +105,20 @@ export class AuthService {
      return res;
    });
  }
+
+ // Send email to user for password reset if they forget password
```
Comment (addarsh-google) [inline src/main/ngapp/src/app/auth/auth.service.ts:109]: nit: formatting

File: src/main/ngapp/src/app/login/login.component.ts [hunk 3]
```diff
@@ -11,6 +11,8 @@ export class LoginComponent implements OnInit {
  loginAwait = false;
  error: any;
  errorPresent = false;
+ passwordNotRight = false; // Tracks if a login was successful
```
Comment (addarsh-google) [inline src/main/ngapp/src/app/login/login.component.ts:14]: rename to loginSuccessful = true?

File: src/main/ngapp/src/app/login/login.component.ts [hunk 4]
```diff
@@ -53,4 +56,12 @@ export class LoginComponent implements OnInit {
        this.error = error.message;
      });
  }
+
+ sendPasswordResetEmail(email: string) {
+    if(!email) {
```
Comment (addarsh-google) [inline src/main/ngapp/src/app/login/login.component.ts:61]: should you be checking for empty string here?

File: src/main/ngapp/src/app/login/login.component.ts (hunk 3 above)
Comment (kiprot) [reply src/main/ngapp/src/app/login/login.component.ts:14]: alright. 
implemented.

Comment (addarsh-google) [reply src/main/ngapp/src/app/login/login.component.ts:14]: did you send out new commits? Can't see one.

Comment (kiprot) [reply src/main/ngapp/src/app/login/login.component.ts]: new commits sent ( i had been waiting for reply to the comment on line 62 before making a commit)

File: src/main/ngapp/src/app/login/login.component.ts (hunk 4 above)
Comment (kiprot) [reply src/main/ngapp/src/app/login/login.component.ts:61]: i was checking to see if the email field had been filled by user before sending email.
is there a better way/ place to do this?

Comment (addarsh-google) [reply src/main/ngapp/src/app/login/login.component.ts:61]: No the place is fine. I was just wondering if the condition should be " email === '' "?

Comment (kiprot) [reply src/main/ngapp/src/app/login/login.component.ts:61]: i think it should, actually. i'll make the change

**Pull request merged**